waspc/cli/src/Wasp/Cli/Command/CreateNewProject.hs

```diff
@@ -1,40 +1,41 @@
-{-# LANGUAGE ViewPatterns #-}
-
 module Wasp.Cli.Command.CreateNewProject
   ( createNewProject,
   )
 where
 
 import Control.Monad.Except (throwError)
 import Control.Monad.IO.Class (liftIO)
-import Data.Char (isLetter)
 import Data.List (intercalate)
 import StrongPath (Abs, Dir, File', Path', Rel, reldir, relfile, (</>))
 import qualified StrongPath as SP
 import System.Directory (createDirectory, getCurrentDirectory)
 import qualified System.Directory
 import qualified System.FilePath as FP
 import Text.Printf (printf)
+import Wasp.Analyzer.Parser (isValidWaspIdentifier)
 import Wasp.AppSpec.ExternalCode (SourceExternalCodeDir)
 import Wasp.Cli.Command (Command, CommandError (..))
 import qualified Wasp.Cli.Command.Common as Command.Common
 import qualified Wasp.Cli.Common as Common
 import qualified Wasp.Data
-import Wasp.Lexer (reservedNames)
+import Wasp.Util (indent)
 import qualified Wasp.Util.Terminal as Term
 
 newtype ProjectName = ProjectName {_projectName :: String}
 
 createNewProject :: String -> Command ()
-createNewProject (all isLetter -> False) =
-  throwError $ CommandError "Project creation failed" "Please use only letters for a new project's name."
-createNewProject ((`elem` reservedNames) -> True) =
-  throwError $
-    CommandError
-      "Project creation failed"
-      ( "Please pick a project name not one of these reserved words:\n\t" ++ intercalate "\n\t" reservedNames
-      )
-createNewProject name = createNewProject' (ProjectName name)
+createNewProject name
```

Comment:
Yes, that's definitely a better way. I forgot that we no longer have the third guard.

Reply:
> you were asking if this piece of code can be simplified by using monadic approach -> what did you have in mind? I don't see at first any idea on how to approach it that way. From what I see, we are throwing an error that we construct by concatenating strings and that is it.

It's hard to articulate because I didn't really have anything in mind. Something just seemed off. I'll make a note of this and come back to it in a few months when my Haskell gets better.

Reply:
I think I do get the intuition you are having, it is connected with getting True/False and then acting on those, it feels like you have Just/Nothing and then can connect stuff monadically, but Boolean makes that hard and I believe if you try anything it will just become more complicated. But I am also not an expert so who knows, would be interesting to see what you figure out in couple of months!